src/cargo/ops/registry.rs

```diff
@@ -442,13 +443,26 @@ fn wait_for_publish(
 ) -> CargoResult<()> {
     let version_req = format!("={}", pkg.version());
     let mut source = SourceConfigMap::empty(config)?.load(registry_src, &HashSet::new())?;
-    let source_description = source.describe();
+    // Disable the source's built-in progress bars. Repeatedly showing a bunch
+    // of independent progress bars can be a little confusing. There is an
+    // overall progress bar managed here.
+    source.set_quiet(true);
+    let source_description = source.source_id().to_string();
     let query = Dependency::parse(pkg.name(), Some(&version_req), registry_src)?;
 
     let now = std::time::Instant::now();
     let sleep_time = std::time::Duration::from_secs(1);
-    let mut logged = false;
-    loop {
+    let max = timeout.as_secs() as usize;
+    config.shell().status("Published", pkg.to_string())?;
+    // Short does not include the registry name.
+    let short_pkg_description = format!("{} v{}", pkg.name(), pkg.version());
+    config.shell().note(format!(
+        "Waiting up to {max} seconds for `{short_pkg_description}` to be available at {source_description}.\n\
+        You may press ctrl-c to skip waiting; the crate should be available shortly."
+    ))?;
```

Comment:
When looking at the output, I felt this note was too busy

Reply:
Do you have any suggestions to improve it? I don't see any way to shorten the first sentence any more than it is, and I think it is important to inform the user what Cargo is about to do. The second sentence I think is pretty awkward, but I couldn't come up with anything much better. I think the previous text of `(ctrl-c to wait asynchronously)` was short and to-the-point, but not very clear to me on what that meant.

Reply:
I dropped the "up to 60 seconds" part to de-clutter since it didn't seem too terribly important for the user to know the exact value.